components/arm/fake/fake.go

```diff
@@ -67,8 +68,8 @@ func (config *AttrConfig) Validate(path string) error {
 
 func init() {
 	registry.RegisterComponent(arm.Subtype, ModelName, registry.Component{
-		Constructor: func(ctx context.Context, _ registry.Dependencies, config config.Component, logger golog.Logger) (interface{}, error) {
-			return NewArm(config, logger)
+		RobotConstructor: func(ctx context.Context, r robot.Robot, config config.Component, logger golog.Logger) (interface{}, error) {
```

Comment:
I wanna call out that the `RobotConstructor` is a legacy constructor.
we should move away from using this....
_highlights_ the importance of 
https://viam.atlassian.net/browse/RSDK-1229

lmk your thoughts